python/ray/tune/checkpoint_manager.py

```diff
@@ -114,6 +118,11 @@ def replace_newest_memory_checkpoint(self, new_checkpoint):
         self._newest_memory_checkpoint = new_checkpoint
 
     def on_checkpoint(self, checkpoint):
+        self._cur_order += 1
+        checkpoint.order = self._cur_order
+        self.on_checkpoint_internal(checkpoint)
+
+    def on_checkpoint_internal(self, checkpoint):
```

Comment:
Do we need the `on_checkpoint_internal` function here? Seems like we never call it anywhere else

Reply:
oh right, making it private

Reply:
What's the purpose of splitting out `on_checkpoint_internal` into a separate method altogether? 

What logic should be contained in the external method vs. the internal method?

Reply:
Yep, would also opt to remove the function completely and just keep `on_checkpoint`

Reply:
sure. combining the two.